Gems/ScriptAutomation/Code/Source/ScriptAutomationScriptBindings.cpp

```diff
@@ -355,16 +362,22 @@ namespace ScriptAutomation
                 // Note this will pause the script until the capture is complete
                 if (PrepareForScreenCapture(imageName))
                 {
-                    AZStd::string screenshotFilePath;
+                    AZ::Outcome<AZStd::string, AZ::Render::FrameCaptureTestError> pathOutcome;
                     AZ::Render::FrameCaptureTestRequestBus::BroadcastResult(
-                        screenshotFilePath, &AZ::Render::FrameCaptureTestRequestBus::Events::BuildScreenshotFilePath, imageName, true);
+                        pathOutcome, &AZ::Render::FrameCaptureTestRequestBus::Events::BuildScreenshotFilePath, imageName, true);
+
+                    AZ_Assert(pathOutcome.IsSuccess(), "Path check should already be done in PrepareForScreenCapture().");
+                    AZStd::string screenshotFilePath = pathOutcome.GetValue();
 
                     auto scriptAutomationInterface = ScriptAutomationInterface::Get();
-                    AZ::Render::FrameCaptureId frameCaptureId = AZ::Render::InvalidFrameCaptureId;
-                    AZ::Render::FrameCaptureRequestBus::BroadcastResult(frameCaptureId, &AZ::Render::FrameCaptureRequestBus::Events::CaptureScreenshot, screenshotFilePath);
-                    if (frameCaptureId != AZ::Render::InvalidFrameCaptureId)
+                    AZ::Outcome<AZ::Render::FrameCaptureId, AZ::Render::FrameCaptureError> captureOutcome;
+                    AZ::Render::FrameCaptureRequestBus::BroadcastResult(captureOutcome, &AZ::Render::FrameCaptureRequestBus::Events::CaptureScreenshot, screenshotFilePath);
+
+                    AZ_Error("ScriptAutomation", captureOutcome.IsSuccess(),
+                        "Frame capture initialization failed. %s", captureOutcome.GetError().m_errorMessage.c_str());
+                    if (captureOutcome.IsSuccess())
                     {
-                        scriptAutomationInterface->SetFrameCaptureId(frameCaptureId);
+                        scriptAutomationInterface->SetFrameCaptureId(captureOutcome.GetValue());
```

Comment:
nit: lots of duplicated code in `CaptureScreenshot`, `CaptureScreenshotWithPreview`, and `CapturePassAttachment`, can any of it be moved to PrepareForScreenCapture?

Reply:
Since I am merging script reporter and will touch this code, I'll simplify this in the next round.